scripts/backup/backup

```diff
@@ -0,0 +1,104 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+UMBREL_ROOT="$(readlink -f $(dirname "${BASH_SOURCE[0]}")/../..)"
+BACKUP_FOLDER="backup"
+BACKUP_ROOT="${UMBREL_ROOT}/${BACKUP_FOLDER}"
+BACKUP_FILE="${UMBREL_ROOT}/backup.tar.gz.pgp"
+
+# Deterministically derives 128 bits of cryptographically secure entropy
+derive_entropy () {
+  identifier="${1}"
+  umbrel_seed=$(cat "${UMBREL_ROOT}/db/umbrel-seed/seed")
+
+  if [[ -z "$umbrel_seed" ]] || [[ -z "$identifier" ]]; then
+    echo "Missing derivation parameter, this is unsafe, exiting."
+    exit 1
+  fi
+
+  # We need `sed 's/^.* //'` to trim the "(stdin)= " prefix from some versions of openssl
+  echo -n "${identifier}" | openssl dgst -sha256 -hmac "${umbrel_seed}" | sed 's/^.* //'
+}
+
+# Make sure an update is not in progres
+if [[ -f "${UMBREL_ROOT}/statuses/backup-in-progress" ]]; then
+    echo "A backup is already in progress. Exiting now."
+    exit 1
+fi
+
+echo "Creating lock..."
+touch "${UMBREL_ROOT}/statuses/backup-in-progress"
+
+[[ -f "${UMBREL_ROOT}/.env" ]] && source "${UMBREL_ROOT}/.env"
+BITCOIN_NETWORK=${BITCOIN_NETWORK:-mainnet}
+
+[[ -d "${BACKUP_ROOT}" ]] && rm -rf "${BACKUP_ROOT}"
+
+echo "Deriving keys..."
+
+backup_id=$(derive_entropy "umbrel_backup_id")
+encryption_key=$(derive_entropy "umbrel_backup_encryption_key")
+
+echo "Creating backup..."
+
+mkdir -p "${BACKUP_ROOT}"
+
+cp --archive "${UMBREL_ROOT}/lnd/data/chain/bitcoin/${BITCOIN_NETWORK}/channel.backup" "${BACKUP_ROOT}/channel.backup"
+cp --archive "${UMBREL_ROOT}/db/user.json" "${BACKUP_ROOT}/user.json"
+
+echo "Adding random padding..."
+
+# Up to 10KB of random binary data
+padding="$(shuf -i 0-10240 -n 1)"
+dd if=/dev/urandom bs="${padding}" count=1 > "${BACKUP_ROOT}/.padding"
```

Comment:
can we please call it `.poop` instead of `.padding`???????

Reply:
"we add a little bit of shit to your backup to prevent a time analysis attack" lmao

Reply:
I agree.,..